homeassistant/components/withings/common.py

```diff
@@ -770,8 +771,16 @@ async def async_get_measures(self) -> Dict[MeasureType, Any]:
 
         response = await self._hass.async_add_executor_job(self._api.measure_get_meas)
 
-        groups = query_measure_groups(
-            response, MeasureTypes.ANY, MeasureGroupAttribs.UNAMBIGUOUS
+        def get_sort_key(group: MeasureGetMeasGroup) -> str:
+            return group.created.datetime
+
+        # Sort from oldest to newest.
+        groups = sorted(
+            query_measure_groups(
+                response, MeasureTypes.ANY, MeasureGroupAttribs.UNAMBIGUOUS
+            ),
+            key=get_sort_key,
```

Comment:
Lambdas are perfect for this too :)

```suggestion
            key=lambda group: group.created.datetime,
```